### DISCUSSION TITLE
Re-enabling disabled spells in Serpent Isle?

### DISCUSSION THREAD

**Discussion by GrantWChapman** [exult/exult#549] [General]
Has anyone had any luck getting blink, fire snake, and time stop to work in Serpent Isle?
I know blink was disabled because it could get you past plot triggers and a locked door or two, but I'd still like to see them working again if possible.
It appears that the spells work properly in the SI Beta, would it be possible to overwrite the code for the spells in the final version with their beta counterparts? If so what would need to be done to do so?

**Comment by DominusExult** [comment]
SI Fixes is the way to go ;)
Firesnake spell is working there and I had no idea that Time Stop does not work 🤷‍♂️

**Reply by GrantWChapman** [reply]
Does SI Fixes have those spells working? I thought they were still broken. I'll have to give them a try.

**Reply by GrantWChapman** [reply]
It looks like firesnake is working in SI Fixes, but time stop and blink both still fail.

**Reply by DominusExult** [reply]
Yes, I knew only firesnake worked. I had no idea time stop doesn‘t work (can you refresh my memory what it is supposed to do?).
Blink spell is rightly considered too dangerous 🤪
(But if you want a mod is the way to do it)

**Reply by GrantWChapman** [reply]
Time stop is supposed to be a 20 second pause on ai and time. Even if blink causes problems it feels wrong to have a completely useless spell in the spellbook, either that or blink could be replaced with another spell.

**Reply by KnightCaptainU7** [reply]
Blink is intentionally broken as a symptom of the imbalance plaguing the land. Or the game testers finding places where it could break the plot.

```
void circle4Blink object#(0x658) ()
{
	var failure;

	if (event == DOUBLECLICK)
	{
		UI_halt_scheduled(item);
		UI_item_say(item, "@Rel Por@");
		failure = UI_execute_usecode_array(item, [no_halt, npc_frame_swing1, npc_frame_swing2h2, npc_frame_swing1, call_usecode, SCRIPT_SPELL_FAILED]);
	}
}
```

**Reply by KnightCaptainU7** [reply]
Doesn't explain why Selina's blink ring works for her but not for you. Some protection from Batlin or The Guardian? Evil humanoids less impacted?

**Reply by DominusExult** [reply]
I'd say it works because the spell has previously been casted on the ring and it just releases it.
Casting it is what is failing due to imbalance 🤷‍♂️

**Comment by MithranArkanere** [comment]
Rather than replacing the spell, I would go with a change in the behavior, prompted by the storms, indicated with a magic lightning effect. 

For example, rather than teleporting you, Blink would teleport all but one enemy farther away, making it easier to kill them one by one, summon some random enemy, or transform them into something else like a weaker monster or food.  
Or a variety of random effects, similar to the random storm effects, indicated by the color of the lightning: Red teleports the enemy away. Yellow replaces them with something, like food or reagents. Green turns them into a different monster.

If it was possible to check if a tile is on a region of the screen, it would also be possible to ensure the enemy isn't teleported to an unreachable location by never triggering the teleport effect while a wall tile is within the teleporting range of the spell.

Another possibility would be determining a viable location by casting an invisible projectile can only pass through certain tiles, teleporting where the projectile either reaches its maximum range or a tile it cannot cross.
